File: increasingring/src/main/java/com/baybaka/increasingring/service/ServiceStarter.java
package com.baybaka.increasingring.service;

import android.content.Context;
import android.content.Intent;

import com.baybaka.increasingring.Getter;
import com.baybaka.increasingring.SettingsService;

import org.slf4j.LoggerFactory;

public class ServiceStarter {

    public static void startServiceWithCondition(Context context) {
        if (checkIsServiceEnabledInConfig(context)) {
            context.startService(new Intent(context, VolumeService.class));
        }
    }

    private static boolean checkIsServiceEnabledInConfig(Context context) {
        SettingsService settingsService = ((Getter)context.getApplicationContext()).getSetting();
        boolean result = settingsService.isServiceEnabledInConfig();
        LoggerFactory.getLogger(ServiceStarter.class.getSimpleName()).info("Service config status is {}: ", result);
        return result;
    }

    public static void stopServiceRestartIfEnabled(Context context) {
        LoggerFactory.getLogger(ServiceStarter.class.getSimpleName()).info("Stopping service");
        context.stopService(new Intent(context, VolumeService.class));
        startServiceWithCondition(context);
    }
}

File: increasingring/src/main/java/com/baybaka/increasingring/service/ServiceStarter.kt
package com.baybaka.increasingring.service

import android.content.Context
import android.content.Intent
import com.baybaka.increasingring.di.Getter
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object ServiceStarter {
    fun logger(): Logger = LoggerFactory.getLogger(ServiceStarter::class.java.simpleName)

    fun startServiceWithCondition(context: Context) {
        if (checkIsServiceEnabledInConfig(context)) {
            context.startService(Intent(context, VolumeService::class.java))
        }
    }

    private fun checkIsServiceEnabledInConfig(context: Context): Boolean {
        val settingsService = (context.applicationContext as Getter).setting
        val result = settingsService.isServiceEnabledInConfig
        logger().info("Service config status is {}: ", result)
        return result
    }

    fun stopServiceRestartIfEnabled(context: Context) {
        logger().info("Stopping service")
        context.stopService(Intent(context, VolumeService::class.java))
        startServiceWithCondition(context)
    }
}
